frontend/src/app.js

```diff
@@ -20,6 +22,8 @@ function App() {
           {isSignedIn && <Route path='/user' element={<User />} />}{' '}
           <Route path='/books' element={<ViewAllBooks />} />
           <Route path='/books/:book_id' element={<ViewBook />} />
+          {isSignedIn && <Route path='/collection/:user_id' element={<ViewCollection />} />}{' '}
+          <Route path='/collection/:user_id/:book_id' element={<AddBook />} />
```

Comment:
i think this path should be private or otherwise, the AddBook page should handle the case where user is not signed in